cpp/src/cylon/indexing/index.cpp

```diff
@@ -0,0 +1,28 @@
+#include "index.hpp"
+
+namespace cylon {
+
+std::unique_ptr<IndexKernel> CreateHashIndexKernel(std::shared_ptr<arrow::Table> input_table, int index_column) {
+  switch (input_table->column(index_column)->chunk(0)->type()->id()) {
+
+    case arrow::Type::NA:return nullptr;
```

Comment:
shall we implement the others as well?

Reply:
Yes, this is just a very initial functional draft. Just want to point out the rough workflow.